src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -48,8 +48,12 @@ struct OperationParameters {
 
             switch httpParam.in {
             case .query:
-                viewModel.optional ? query.optional.append(viewModel) : query.required
-                    .append(viewModel)
+                if param.implementation == .client {
+                    query.optionalClientProperty.append(viewModel)
+                } else {
+                    viewModel.optional ? query.optional.append(viewModel) : query.required
+                        .append(viewModel)
+                }
```

Comment:
So right now it seems that `query` has two properties, `optionalClientProperty` and `optional`. Would it make more sense for `query.optional` to have a property that described where it comes from? `querty.optional.source` or something like that? I just worry that if there is another implementation location in the future, then we have ever-increasing `optionalFoo` properties.

Reply:
I add a 'source' property in the KeyValue Model to indicate where the value of the KeyValueModel comes from.   And the 'source' property will be used in OperationOptionsSnippet and OperationSnippet.stencil so that we can unwrap optional options/client property before adding it to query parameter of the URL.